File: jecnaapi-java/src/examples/java/GradesExample.java
import io.github.tomhula.jecnaapi.data.grade.Grade;
import io.github.tomhula.jecnaapi.data.grade.GradesPage;
import io.github.tomhula.jecnaapi.data.grade.Subject;
import io.github.tomhula.jecnaapi.data.grade.WeightedGradeAverageCalculator;
import io.github.tomhula.jecnaapi.java.JecnaClientJavaWrapper;
import io.github.tomhula.jecnaapi.util.SchoolYear;
import io.github.tomhula.jecnaapi.util.SchoolYearHalf;
import io.github.tomhula.jecnaapi.web.Auth;

import java.util.List;

public class GradesExample
{
    public static void main(String[] args)
    {
        /* Metody join() používáme aby jsme z asynchroního běhu "udělali" synchroní; pouze v příkladech */

        JecnaClientJavaWrapper jecnaClient = new JecnaClientJavaWrapper();

        jecnaClient.login("user", "password").join();

        /* Stáhne známky z roku 2021/2022 z druhého pololetí. */
        GradesPage gradesPage = jecnaClient.getGradesPage(new SchoolYear(2021), SchoolYearHalf.SECOND).join();

        Subject mathSubject = gradesPage.get("Matematika");

        /* Průmer známek */
        float mathAverage = mathSubject.getGrades().average();

        System.out.println("Průměr z matematiky: " + mathAverage);

        /* Do .get() patří string, který určuje část předmětu. (např. Teorie, Cvičení)
         * Pokud předmět není rozdělen, použijte null. */
        List<Grade> mathGrades = mathSubject.getGrades().get(null);

        System.out.println("Známky z matematiky: ");

        for (Grade grade : mathGrades)
            System.out.println(grade);
    }
}

File: src/examples/kotlin/GradesExample.kt
import kotlinx.coroutines.runBlocking
import io.github.tomhula.jecnaapi.JecnaClient
import io.github.tomhula.jecnaapi.util.SchoolYear
import io.github.tomhula.jecnaapi.util.SchoolYearHalf

fun main(): Unit = runBlocking {

    val client = JecnaClient()

    client.login("user", "password")

    /* Stáhne známky z roku 2021/2022 z druhého pololetí. */
    val gradesPage = client.getGradesPage(SchoolYear(2021), SchoolYearHalf.SECOND)

    val mathSubject = gradesPage["Matematika"]!!

    /* Průmer známek */
    val mathAverage = mathSubject.grades.average()

    println("Průměr z matematiky: $mathAverage")

    /* Do grades[] patří string, který určuje část předmětu. (např. Teorie, Cvičení)
    * Pokud předmět není rozdělen, použijte null. */
    val mathGrades = mathSubject.grades[null]!!

    println("Známky z matematiky: ")

    for (grade in mathGrades)
        println(grade.valueChar() + " - " + grade.description)
}
